TestProjects/PerceptionURP/Assets/ExampleScripts/CustomAnnotationAndMetricReporter.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.Perception.GroundTruth;
+
+[RequireComponent(typeof(PerceptionCamera))]
+public class CustomAnnotationAndMetricReporter : MonoBehaviour
+{
+    public GameObject light;
+    public GameObject target;
+
+    MetricDefinition lightMetricDefinition;
+    AnnotationDefinition boundingBoxAnnotationDefinition;
+    SensorHandle cameraSensorHandle;
+
+    public void Start()
+    {
+        //Metrics and annotations are registered up-front
+        lightMetricDefinition = SimulationManager.RegisterMetricDefinition(
+            "Light position",
+            "The world-space position of the light",
+            Guid.Parse("1F6BFF46-F884-4CC5-A878-DB987278FE35"));
+        boundingBoxAnnotationDefinition = SimulationManager.RegisterAnnotationDefinition(
+            "Target bounding box",
+            "The position of the target in the camera's local space",
+            id: Guid.Parse("C0B4A22C-0420-4D9F-BAFC-954B8F7B35A7"));
+    }
+
+    public void Update()
+    {
+        //Report the light's position by manually creating the json array string.
+        var lightPos = light.transform.position;
+        SimulationManager.ReportMetric(lightMetricDefinition,
+            $@"[{{ ""x"": {lightPos.x}, ""y"": {lightPos.y}, ""z"": {lightPos.z} }}]");
+        //compute the location of the object in the camera's local space
+        Vector3 targetPos = transform.worldToLocalMatrix * target.transform.position;
+        //Report using the PerceptionCamera's SensorHandle if scheduled this frame
+        var sensorHandle = GetComponent<PerceptionCamera>().SensorHandle;
+        if (sensorHandle.ShouldCaptureThisFrame)
+        {
+            sensorHandle.ReportAnnotationValues(
+                boundingBoxAnnotationDefinition,
+                new[] { targetPos });
+        }
+    }
+}
+
+// Example metric that is added each frame in the dataset:
```

Comment:
Maybe we should remove the comments from the script and leave a comment to the file that contains the example of this script

Reply:
I was following the Microsoft pattern for C# examples https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1?view=netcore-3.1
You think it's too long?

Reply:
probably ok since it is in a example scripts folder